File: app/src/main/java/com/cool/eye/mvp/java/IPresenter.java
package com.cool.eye.mvp.java;

/**
 * @author ycb
 * @date 2018/8/17
 */
public interface IPresenter<V extends IView> {

    V getView();

    void onCreate();

    void onStart();

    void onPause();

    void onResume();

    void onStop();

    void onDestroy();

    void onRefresh();

    void onLoadMore();

    void onRestart();
}

File: app/src/main/java/com/cool/eye/mvp/kotlin/IPresenter.kt
package com.cool.eye.mvp.kotlin

/**
 * @author ycb
 * @date 2018/8/17
 */
interface IPresenter<V : IView> {

    val view: V

    fun onCreate() {}

    fun onStart() {}

    fun onPause() {}

    fun onResume() {}

    fun onStop() {}

    fun onDestroy() {}

    fun onRefresh() {}

    fun onLoadMore() {}

    fun onRestart() {}
}
